From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Rochelemagne <rochelemagne@cpanel.net>
Date: Sat, 10 Mar 2018 10:43:20 -0600
Subject: [PATCH] Storable do not load Fcntl

Case CPANEL-19113:

Patch Storable to mitigate issue with recent SEGV
---
 SOURCES/perl/dist/Storable/Storable.pm | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/SOURCES/perl/dist/Storable/Storable.pm b/SOURCES/perl/dist/Storable/Storable.pm
index ef417c69..c8a39ec6 100644
--- a/SOURCES/perl/dist/Storable/Storable.pm
+++ b/SOURCES/perl/dist/Storable/Storable.pm
@@ -80,16 +80,9 @@ BEGIN {
 # They might miss :flock in Fcntl
 #
 
-BEGIN {
-    if (eval { require Fcntl; 1 } && exists $Fcntl::EXPORT_TAGS{'flock'}) {
-        Fcntl->import(':flock');
-    } else {
-        eval q{
-	          sub LOCK_SH () { 1 }
-		  sub LOCK_EX () { 2 }
-	      };
-    }
-}
+# avoid to loading Fcntl
+sub LOCK_SH () { 1 }
+sub LOCK_EX () { 2 }
 
 sub CLONE {
     # clone context under threads
